src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -527,18 +529,65 @@ public String getLaunchURLForCaseId(final UUID caseId, LaunchRequestDTO requestP
 
     CaseContainerDTO caseDetails = getLaunchCase(caseId);
 
+    SingleUseQuestionnaireIdDTO newQuestionnaireIdDto =
+        getNewQidForCase(caseDetails, requestParamsDTO.getIndividual());
+
+    String questionnaireId = newQuestionnaireIdDto.getQuestionnaireId();
+    String formType = newQuestionnaireIdDto.getFormType();
+
+    String eqUrl = createLaunchUrl(formType, caseDetails, requestParamsDTO, questionnaireId);
+    publishSurveyLaunchedEvent(caseDetails.getId(), questionnaireId, requestParamsDTO.getAgentId());
+    return eqUrl;
+  }
+
+  @Override
+  public UACResponseDTO getUACForCaseId(UUID caseId, UACRequestDTO requestParamsDTO)
+      throws CTPException {
+    log.with("caseId", caseId)
+        .with("request", requestParamsDTO)
+        .debug("Processing request to get UAC for Case");
+
+    CaseContainerDTO caseDetails = getLaunchCase(caseId);
+
+    SingleUseQuestionnaireIdDTO newQuestionnaireIdDto =
+        getNewQidForCase(caseDetails, requestParamsDTO.getIndividual());
+
+    UACResponseDTO response =
```

Comment:
Do you need the reponse as a field or can't you just return UACResponseDTO.builder().. etc?

Reply:
As it's you Andy - done.